### PR TITLE
Feature network architecture c++ http client frontend + server architecture expressjs

### PR DESCRIPTION
**Pull request by theophile-jr**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,6 @@
 *.exe
 *.out
 *.app
+
+#for nodejs
+node_modules/
```
Comment (AxelF44) [inline .gitignore]:
```suggestion
node_modules/

```

Comment (Marius-P1) [inline .gitignore:35]:
```suggestion
node_modules/

```

File: Makefile [hunk 2]
```diff
@@ -46,3 +52,5 @@ re:	fclean all
 	@g++ -c $< -o $@ $(CXXFLAGS) $(INCLUDE) && \
 	$(call YELLOW,"🆗 $<") || \
 	$(call YELLOW,"❌ $<")
+
+.PHONY: server
```
Comment (AxelF44) [inline Makefile]:
```suggestion
.PHONY: server

```

Comment (Marius-P1) [inline Makefile:56]:
```suggestion
.PHONY: server

```

File: src/main.cpp [hunk 3]
```diff
@@ -8,12 +8,15 @@
 #include <iostream>
 
 #include "Renderer.hpp"
+#include "Network.hpp"
```
Comment (AxelF44) [inline src/main.cpp:11]:
```suggestion
#include "Network.hpp"
#include "Renderer.hpp"
```

File: include/Network.hpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/*
+** EPITECH PROJECT, 2024
+** FlameClicker
+** File description:
+** Network.hpp
+*/
+
+#ifndef NETWORK_HPP_
```
Comment (Marius-P1) [inline include/Network.hpp:8]: You can use pragma once

File: src/main.cpp [hunk 5]
```diff
@@ -8,12 +8,15 @@
...
 
 int main(void)
 {
     try {
-        FlameClicker::Renderer renderer;
-        renderer.renderClicker();
+        // FlameClicker::Renderer renderer;
+        // renderer.renderClicker();
```
Comment (AxelF44) [inline src/main.cpp:17]:
```suggestion
```

File: include/Network.hpp [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#define NETWORK_HPP_
+
+#define API_URL_DEV "http://localhost:3000"
+#define API_URL_PROD "not defined yet"
+
+#include "json-parser/simdjson.h"
+
+namespace Network {
+    std::string requestEndpoint(std::string endpoint = "/", bool isProd = false);
```
Comment (mathieurobert1) [inline include/Network.hpp:27]: comment

File: include/Network.hpp [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    std::string getJSONSTRINGparameter(std::string json, std::string parameter);
```
Comment (mathieurobert1) [inline include/Network.hpp:36]: comment

File: src/network/Request.cpp [hunk 8]
```diff
@@ -0,0 +1,35 @@
+#include <curl/curl.h>
+#include <iostream>
```
Comment (AxelF44) [inline src/network/Request.cpp:2]:
```suggestion
#include <iostream>
#include <curl/curl.h>
```

File: src/network/Request.cpp [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "json-parser/simdjson.h"
+#include "Network.hpp"
```
Comment (AxelF44) [inline src/network/Request.cpp:5]:
```suggestion
#include "Network.hpp"
#include "json-parser/simdjson.h"
```

File: include/Network.hpp [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/*
+** EPITECH PROJECT, 2024
+** FlameClicker
+** File description:
+** Network.hpp
+*/
+
+#ifndef NETWORK_HPP_
+#define NETWORK_HPP_
```
Comment (Njord201) [inline include/Network.hpp:9]: Use pragma once

File: include/Network.hpp [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+#define API_URL_PROD "not defined yet"
+
+#include "json-parser/simdjson.h"
+
+namespace Network {
+    std::string requestEndpoint(std::string endpoint = "/", bool isProd = false);
+    std::string getJSONSTRINGparameter(std::string json, std::string parameter);
+    class Api {
+
+    };
```
Comment (Njord201) [inline include/Network.hpp:21]: Use doxygen

File: server/index.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (Njord201) [inline server/index.js:1]: Header

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by Marius-P1 — DISMISSED** [review]
:ok_hand:

**Review by mathieurobert1 — DISMISSED** [review]

**Review by Njord201 — COMMENTED** [review]
Do request of @AxelF44

File: src/main.cpp (hunk 3 above)
Comment (theophile-jr) [reply src/main.cpp:11]: Because renderer is longer of course :')

**Review by mathieurobert1 — APPROVED** [review]

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by AxelF44 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Pull request merged**